library/core/tests/num/int_macros.rs

```diff
@@ -290,6 +290,34 @@ macro_rules! int_module {
                 assert_eq!(r.saturating_pow(0), 1 as $T);
             }
 
+            #[test]
+            fn test_isqrt() {
+                assert_eq!($T::MIN.checked_isqrt(), None);
+                assert_eq!((-1 as $T).checked_isqrt(), None);
+                assert_eq!((0 as $T).isqrt(), 0 as $T);
+                assert_eq!((1 as $T).isqrt(), 1 as $T);
+                assert_eq!((2 as $T).isqrt(), 1 as $T);
+                assert_eq!((99 as $T).isqrt(), 9 as $T);
+                assert_eq!((100 as $T).isqrt(), 10 as $T);
+
+                let n_max: $T = (1024 * 1024).min($T::MAX as u128) as $T;
+                for n in 0..=n_max {
```

Comment:
This needs a `cfg(not(miri))` like in:

https://github.com/rust-lang/rust/blob/c01d8d238c57f9847e1f1dd5a1d2659b85e2a468/library/core/tests/num/int_log.rs#L96-L103

```suggestion
            }

            #[cfg(not(miri))] // Miri is too slow.
            #[test]
            fn test_lots_of_isqrt() {
                let n_max: $T = (1024 * 1024).min($T::MAX as u128) as $T;
                for n in 0..=n_max {
```

Reply:
Right! I changed it also in [`uint_macros.rs`](https://github.com/rust-lang/rust/pull/116176/files#diff-16e4c12b3c98db6567d22afdaca11b5058848325709e24f1b14097cd731fc071R304-R323).